File: src/Models/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

/**
 *
 * @author user
 */
public class Produit {
    int id ;
    String nom;
    double prix;
    int Qte;
    int idCategorie;
    
        public Produit(int id, String nom, double prix, int Qte, int idCategorie) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.Qte = Qte;
        this.idCategorie = idCategorie;
    }

    public Produit(String nom, double prix, int Qte, int idCategorie) {
        this.nom = nom;
        this.prix = prix;
        this.Qte = Qte;
        this.idCategorie = idCategorie;
    }

    public Produit() {
    }


    public Produit(String nom, double prix, int idCategorie) {
        this.nom = nom;
        this.prix = prix;
        this.idCategorie = idCategorie;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getQte() {
        return Qte;
    }

    public void setQte(int Qte) {
        this.Qte = Qte;
    }

    public int getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    @Override
    public String toString() {
        return "Produit{" + "id=" + id + ", nom=" + nom + ", prix=" + prix + ", Qte=" + Qte + ", idCategorie=" + idCategorie + '}';
    }
    
    
}

File: Produit.php
<?php

namespace App\Entity;

use App\Repository\ProduitRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=ProduitRepository::class)
 */
class Produit
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     *@Assert\NotBlank(message="valider")

     */
    private $nom;

    /**
     * @ORM\Column(type="float")
     *@Assert\NotBlank(message="valider")

     *
     */
    private $prix;

    /**
     * @ORM\Column(type="integer")
     *@Assert\NotBlank(message="valider")
     *@Assert\Length(
     *min=3,
     *max=50,
     *minMessage="Lenom de la qte doit comporter au moins {{ limit }} caractères",
     *maxMessage ="Le nom de la qte doit comporter au plus {{ limit }} caractères"
     *
    )
     */
    private $Qte;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="valider")

     */
    private $idCategorie;

    /**
     * @ORM\Column(type="string", length=255)


     */
    private $image;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getQte(): ?int
    {
        return $this->Qte;
    }

    public function setQte(int $Qte): self
    {
        $this->Qte = $Qte;

        return $this;
    }

    public function getIdCategorie(): ?string
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(string $idCategorie): self
    {
        $this->idCategorie = $idCategorie;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }
}
